### PR TITLE
samples: add i2c-target samples

### PR DESCRIPTION
**Pull request by bravl**
Add two simple i2c target samples

1. custom i2c target: sample how to use the i2c target api
2. i2c eeprom target: sample how to initialize and use the i2c driver target API

### CODE REVIEW

File: samples/drivers/i2c/target_eeprom/README.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+   :name: I2C Target
+   :relevant-api: i2c_interface
+
+   Setup an I2C target on the I2C interface.
+
+Overview
+********
+
+This sample demonstrates how to setup and use the I2C target driver API using the
+`zephyr,i2c-target-eeprom` device.
```
Comment (kartben) [inline samples/drivers/i2c/target_eeprom/README.rst:11]:
```suggestion
:dtcompatible:`zephyr,i2c-target-eeprom` device.
```

File: samples/drivers/i2c/target_eeprom/README.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.. zephyr:code-sample:: i2c-eeprom-target
+   :name: I2C Target
+   :relevant-api: i2c_interface
+
+   Setup an I2C target on the I2C interface.
+
+Overview
+********
+
+This sample demonstrates how to setup and use the I2C target driver API using the
```
Comment (kartben) [inline samples/drivers/i2c/target_eeprom/README.rst:10]:
```suggestion
This sample demonstrates how to setup and use the :ref:`i2c-target-api` using the
```

File: samples/drivers/i2c/custom_target/README.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
+.. zephyr:code-sample:: i2c-custom-target
+   :name: I2C Custom Target
+   :relevant-api: i2c_interface
+
+   Setup a custom I2C target on the I2C interface.
+
+Overview
+********
+
+This sample demonstrates how to setup an I2C custom target on the I2C interface.
```
Comment (kartben) [inline samples/drivers/i2c/custom_target/README.rst:10]:
```suggestion
This sample demonstrates how to setup an I2C custom target on the I2C interface
using the :ref:`i2c-target-api`.
```

**Review by kartben — CHANGES_REQUESTED** [review]
cool, thanks for this! 
Some minor suggested documentation fixes

File: samples/drivers/i2c/target_eeprom/README.rst (hunk 1 above)
Comment (bravl) [reply samples/drivers/i2c/target_eeprom/README.rst]: Fixed!

File: samples/drivers/i2c/target_eeprom/README.rst (hunk 2 above)
Comment (bravl) [reply samples/drivers/i2c/target_eeprom/README.rst]: Fixed!

File: samples/drivers/i2c/custom_target/README.rst (hunk 3 above)
Comment (bravl) [reply samples/drivers/i2c/custom_target/README.rst]: Fixed!

**Review by teburd — DISMISSED** [review]
I like it, thanks for adding this!

**Review by kartben — CHANGES_REQUESTED** [review]
actually, one more request for change: can you please update the MAINTAINERS file to add the new `samples/drivers/i2c` folder to the "Drivers: I2C" entry? This way future updates/additions to the I2C samples will be correctly assigned to @teburd 😛

**Comment by bravl** [comment]
> actually, one more request for change: can you please update the MAINTAINERS file to add the new `samples/drivers/i2c` folder to the "Drivers: I2C" entry? This way future updates/additions to the I2C samples will be correctly assigned to @teburd 😛

added to Maintainers.yml :+1:

**Review by kartben — APPROVED** [review]
Thanks, @bravl!

**Review by teburd — APPROVED** [review]

**Pull request merged**